rasa/shared/core/domain.py

```diff
@@ -1922,38 +1927,87 @@ def validate(mapping: Dict[Text, Any], form_name: Text, slot_name: Text) -> None
         if not isinstance(mapping, dict):
             raise InvalidDomain(
                 f"Please make sure that the slot mappings for slot '{slot_name}' in "
-                f"your form '{form_name}' are valid dictionaries. Please see "
-                f"{DOCS_URL_FORMS} for more information."
+                f"your domain are valid dictionaries."
             )
 
         validations = {
             str(SlotMapping.FROM_ENTITY): ["entity"],
             str(SlotMapping.FROM_INTENT): ["value"],
             str(SlotMapping.FROM_TRIGGER_INTENT): ["value"],
             str(SlotMapping.FROM_TEXT): [],
+            str(SlotMapping.CUSTOM): ["action"],
         }
 
         mapping_type = mapping.get("type")
         required_keys = validations.get(mapping_type)
 
         if required_keys is None:
             raise InvalidDomain(
-                f"Your form '{form_name}' uses an invalid slot mapping of type "
-                f"'{mapping_type}' for slot '{slot_name}'. Please see "
-                f"{DOCS_URL_FORMS} for more information."
+                f"Your slot mapping uses an invalid slot mapping of type "
+                f"'{mapping_type}' for slot '{slot_name}'."
             )
 
         for required_key in required_keys:
             if mapping.get(required_key) is None:
                 raise InvalidDomain(
                     f"You need to specify a value for the key "
                     f"'{required_key}' in the slot mapping of type '{mapping_type}' "
-                    f"for slot '{slot_name}' in the form '{form_name}'. Please see "
-                    f"{DOCS_URL_FORMS} for more information."
+                    f"for slot '{slot_name}'."
                 )
 
+    @staticmethod
+    def intent_is_desired(
+        slot_mapping: Dict[Text, Any], tracker: "DialogueStateTracker",
+    ) -> bool:
+        """Check whether user intent matches intent conditions."""
+        mapping_intents = slot_mapping.get("intent", [])
+        mapping_not_intents = slot_mapping.get("not_intent", [])
+
+        intent = tracker.latest_message.intent.get("name")
+
+        intent_not_blocked = not mapping_intents and intent not in mapping_not_intents
+
+        return intent_not_blocked or intent in mapping_intents
+
+    @staticmethod
+    def entity_is_desired(
```

Comment:
are you actually using this code?

Reply:
Yep, it's used in `ActionExtractSlots.run` - although only once 😄